### PR TITLE
[SPIRV] Tests for the SPIR-V JIT Compiler and Runtime dispatch

### PR DESCRIPTION
**Pull request by jjfumero**
#### Description

This PR provides tests for the SPIR-V JIT Compiler API of TornadoVM and the runtime dispatch logic with Level Zero. 
We already have similar tests for the OpenCL and PTX backends, and the SPIR-V testing was missing.  

This solves issue #202 

#### Backend/s tested

Mark the backends affected by this PR.

- [ ] OpenCL
- [ ] PTX
- [X] SPIRV

#### OS tested

Mark the OS where this PR is tested.

- [X] Linux
- [ ] OSx
- [ ] Windows

#### Did you check on FPGAs?

If it is applicable, check your changes on FPGAs.

- [ ] Yes
- [X] No

#### How to test the new patch?

```bash
$ make BACKEND=spirv
$ test-native.sh 

Testing the Native SPIR-V API

WARNING: Using incubator modules: jdk.incubator.vector, jdk.incubator.foreign
Running Native: uk.ac.manchester.tornado.drivers.spirv.tests.TestSPIRVJITCompilerTask info: s0.t0
	Backend           : SPIRV
	Device            : SPIRV LevelZero - Intel(R) HD Graphics 630 [0x591b] GPU
	Dims              : 1
	Global work offset: [0]
	Global work size  : [128]
	Local  work size  : [128, 1, 1]
	Number of workgroups  : [1]

 ................ [PASS]
WARNING: Using incubator modules: jdk.incubator.foreign, jdk.incubator.vector

Testing the Native OpenCL API

WARNING: Using incubator modules: jdk.incubator.vector, jdk.incubator.foreign
Running Native: uk.ac.manchester.tornado.drivers.opencl.tests.TestOpenCLJITCompiler ................ [PASS]
WARNING: Using incubator modules: jdk.incubator.vector, jdk.incubator.foreign
```
----------------------------------------------------------------------------

### CODE REVIEW

File: drivers/spirv/src/main/java/uk/ac/manchester/tornado/drivers/spirv/graal/SPIRVInstalledCode.java [hunk 1]
```diff
@@ -23,8 +23,16 @@
  */
 package uk.ac.manchester.tornado.drivers.spirv.graal;
 
+import java.io.*;
```
Comment (stratika) [inline drivers/spirv/src/main/java/uk/ac/manchester/tornado/drivers/spirv/graal/SPIRVInstalledCode.java:26]: unfold packages

File: drivers/spirv/src/main/java/uk/ac/manchester/tornado/drivers/spirv/graal/SPIRVLevelZeroInstalledCode.java [hunk 2]
```diff
@@ -34,13 +34,7 @@
...
 import uk.ac.manchester.tornado.drivers.spirv.SPIRVLevelZeroModule;
 import uk.ac.manchester.tornado.drivers.spirv.SPIRVModule;
-import uk.ac.manchester.tornado.drivers.spirv.levelzero.LevelZeroCommandList;
-import uk.ac.manchester.tornado.drivers.spirv.levelzero.LevelZeroKernel;
-import uk.ac.manchester.tornado.drivers.spirv.levelzero.Sizeof;
-import uk.ac.manchester.tornado.drivers.spirv.levelzero.ZeEventHandle;
-import uk.ac.manchester.tornado.drivers.spirv.levelzero.ZeGroupDispatch;
-import uk.ac.manchester.tornado.drivers.spirv.levelzero.ZeKernelHandle;
-import uk.ac.manchester.tornado.drivers.spirv.levelzero.ZeResult;
+import uk.ac.manchester.tornado.drivers.spirv.levelzero.*;
```
Comment (stratika) [inline drivers/spirv/src/main/java/uk/ac/manchester/tornado/drivers/spirv/graal/SPIRVLevelZeroInstalledCode.java:37]: unfold packages

**Review by stratika — CHANGES_REQUESTED** [review]

File: assembly/src/bin/test-native.sh [hunk 3]
```diff
@@ -13,4 +13,18 @@ if [[ $selected_backends == *"opencl"* ]]; then
   tornado uk.ac.manchester.tornado.drivers.opencl.tests.TestOpenCLTornadoCompiler
 fi
 
+if [[ $selected_backends == *"spirv"* ]]; then
+
+  ## The SPIR-V Backend import Level Zero and OpenCL 
+
+  echo -e "\nTesting the Native SPIR-V API\n"
+  tornado --threadInfo uk.ac.manchester.tornado.drivers.spirv.tests.TestSPIRVJITCompiler
```
[the remark this answers is not in the document]
Comment (jjfumero) [reply assembly/src/bin/test-native.sh:21]: We can remove the flag here.

File: drivers/spirv/src/main/java/uk/ac/manchester/tornado/drivers/spirv/SPIRVLevelZeroCodeCache.java [hunk 4]
```diff
@@ -40,17 +40,7 @@
...
-import uk.ac.manchester.tornado.drivers.spirv.levelzero.LevelZeroKernel;
-import uk.ac.manchester.tornado.drivers.spirv.levelzero.LevelZeroModule;
-import uk.ac.manchester.tornado.drivers.spirv.levelzero.ZeBuildLogHandle;
-import uk.ac.manchester.tornado.drivers.spirv.levelzero.ZeKernelDescriptor;
-import uk.ac.manchester.tornado.drivers.spirv.levelzero.ZeKernelHandle;
-import uk.ac.manchester.tornado.drivers.spirv.levelzero.ZeModuleDescriptor;
-import uk.ac.manchester.tornado.drivers.spirv.levelzero.ZeModuleFormat;
-import uk.ac.manchester.tornado.drivers.spirv.levelzero.ZeModuleHandle;
-import uk.ac.manchester.tornado.drivers.spirv.levelzero.ZeResult;
+import uk.ac.manchester.tornado.drivers.spirv.levelzero.*;
```
[the remark this answers is not in the document]
Comment (jjfumero) [reply drivers/spirv/src/main/java/uk/ac/manchester/tornado/drivers/spirv/SPIRVLevelZeroCodeCache.java:43]: m... I think this is due to the code formatter. I will check again.

[the remark this answers is not in the document]
Comment (stratika) [reply drivers/spirv/src/main/java/uk/ac/manchester/tornado/drivers/spirv/SPIRVLevelZeroCodeCache.java:43]: sure, no problem. It has happened to me several times due to misconfiguration of the IntelliJ formatter.

**Comment by jjfumero** [comment]
Fixed. Ok, it was another parameter related to the formatted within IntelliJ. We need to check this documentation.

`Settings -> Java -> Imports`  and then we set a high value (e.g. 100) to the class count for the (`*`).

**Review by stratika — APPROVED** [review]

**Pull request merged**